src/rhsm/config.py

```diff
@@ -124,7 +137,7 @@ def in_container() -> bool:
     ]
     for fn in locations:
         if os.path.exists(fn):
-            return True
+            return not in_ocp()
```

Comment:
Why don't you call `in_ocp()` at the beginning of this method? When result was `True`, then we could return `False` and we would not be any need to check presence of files listed in `locations`.

Reply:
I thought about doing that at first, then I decided against. My reason here is to restrict the loosening (excuse the word play) _only_ when we know this system is a container. I'm not sure whether the situation "has Kubernetes env vars but it is not detected as container" can ever happen - I'd rather deal with it when we actually face it.

Reply:
> I still think that it is not necessary, because we return `False` at the end of this function. If it is not container and these environment variables are set, then we return `False` anyway.

According to your previous suggestion: if I move the `in_ocp()` check before checking for files, then a situation in which those environment variables are set and there are no container files will make `in_container()` return `True`, which is not what I wanted to achieve. To me, having the kubernetes bits ought to be a _specific_ case of "it is a container".

Reply:
OK, but the debug/info print should be present there.